starlite/contrib/sqlalchemy/repository.py

```diff
@@ -0,0 +1,540 @@
+            return (existing, False)
+        return (await self.add(self.model_type(**kwargs)), True)  # type: ignore[arg-type]
+
+    async def count(self, *filters: FilterTypes, **kwargs: Any) -> int:
+        """Get the count of records returned by a query.
+
+        Args:
+            *filters: Types for specific filtering operations.
+            **kwargs: Instance attribute value filters.
+
+        Returns:
+            Count of records returned by query, ignoring pagination.
+        """
+        statement = self.statement.with_only_columns(
+            sql_func.count(  # pylint: disable=not-callable
+                self.model_type.id,  # type:ignore[attr-defined]
+            ),
+            maintain_column_froms=True,
+        ).order_by(None)
+        statement = self._apply_filters(*filters, apply_pagination=False, statement=statement)
+        statement = self._filter_select_by_kwargs(statement, **kwargs)
+        results = await self._execute(statement)
+        return results.scalar_one()  # type: ignore
+
+    async def update(self, data: ModelT) -> ModelT:
+        """Update instance with the attribute values present on `data`.
+
+        Args:
+            data: An instance that should have a value for `self.id_attribute` that exists in the
+                collection.
+
+        Returns:
+            The updated instance.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found with same identifier as `data`.
+        """
+        with wrap_sqlalchemy_exception():
+            item_id = self.get_id_attribute_value(data)
+            # this will raise for not found, and will put the item in the session
+            await self.get(item_id)
+            # this will merge the inbound data to the instance we just put in the session
+            instance = await self._attach_to_session(data, strategy="merge")
+            await self.session.flush()
+            await self.session.refresh(instance)
+            self.session.expunge(instance)
+            return instance
+
+    async def update_many(self, data: list[ModelT]) -> list[ModelT]:
+        """Update one or more instances with the attribute values present on `data`.
+
+        This function has an optimized bulk insert based on the configured SQL dialect:
+        - For backends supporting `RETURNING` with `executemany`, a single bulk insert with returning clause is executed.
+        - For other backends, it does a bulk insert and then selects the inserted records
+
+        Args:
+            data: A list of instances to update.  Each should have a value for `self.id_attribute` that exists in the
+                collection.
+
+        Returns:
+            The updated instances.
+
+        Raises:
+            RepositoryNotFoundException: If no instance found with same identifier as `data`.
+        """
+        data_to_update: list[dict[str, Any]] = [v.to_dict() if isinstance(v, self.model_type) else v for v in data]  # type: ignore
```

Comment:
same comments here as for `add_many`

Reply:
I'm not sure there's an equivalent to `add_all` for bulk updates.  I'm thinking we leave this method as is and then extend this repository further with the `add_many_mapped` and `update_many_mapped` functions.  Thoughts?